src/components/post/index.js

```diff
@@ -1,25 +1,39 @@
 import LikeIcon from "../../assets/icons/likeIcon"
 import CommentIcon from "../../assets/icons/commentIcon"
 import useModal from "../../hooks/useModal"
-import { useState, useEffect } from "react"
+import { useState, useRef } from "react"
 import Card from "../card"
 import Comment from "../comment"
 import EditPostModal from "../editPostModal"
+import DeletePostModal from "../deletePostModal"
 import ProfileCircle from "../profileCircle"
+import useOnOutsideClick from "../../hooks/useOnOutsideClick"
 import "./style.css"
 
-const Post = ({ name, date, content, comments = [], id, likes = 0 }) => {
+const Post = ({ name, date, content, comments = [], id, setIsRerender, likes = 0 }) => {
     const { openModal, setModal } = useModal()
     const [postContent, setContent] = useState(content)
+    const [isMenuVisible, setIsMenuVisible] = useState(false)
+    const menuRef = useRef(null)
 
     const userInitials = name.match(/\b(\w)/g)
 
     const [isLiked, setIsLiked] = useState(false)
 
-    const showModal = () => {
+    useOnOutsideClick(menuRef,()=>setIsMenuVisible(false))
```

Comment:
Looks like you're missing the closing curly bracket

Reply:
They are closed tho?

Reply:
For `showModal`, format this whole file

Reply:
My bad, I can't read today. Format this whole file anyway